test/TestProjects/ExactMatchInheritance/ExactMatchInheritance.json

```diff
@@ -519,6 +598,26 @@
                     "type": "string"
                 }
             }
+        },
+        "AzureResourceFlattenModel": {
+            "type": "object",
+            "description": "This model is x-ms-azure-resource and has flatten properties",
+            "x-ms-azure-resource": true,
+            "properties": {
+                "new": {
```

Comment:
you need to end up with a set of properties that are equal to the set on the reference types.

Couple scenarios i can think of

foo => {bar}
flat1 =>{id, name, type, location, tags, ref:Foo (flatten)} -expect it to inherit from trackedresource<T> with 1 extra property bar
flatreference =>{id, name, type}
flat2 =>{location, tags, bar, ref:flatreference(flatten)} -expect it inherit from trackedresource<T> with 1 extra property bar